### DISCUSSION TITLE
Custom Description for Spontaneous Scopes

### DISCUSSION THREAD

**Discussion by tawaren** [JanssenProject/jans#9749] [Q&A]
When a Spontaneous Scope is requested, the consent page shows the result of the `toString()` method of the corresponding `Scope ` object, which is confusing and uninformative to the end user.

Is there an easy way to set/overwrite the description of a Spontaneous Scope?
If not, do I need to make a custom consent-gathering interception script to achieve a more admirable display?

**Comment by nynymike** [comment]
@yuriyz any thoughts? Could Markus use the Update Token script instead so he doesn't have to implement consent-gathering?

**Reply by yuriyz** [reply]
Currently consent page renders description of the scope as far as I can see:
https://github.com/JanssenProject/jans/blob/23ef80caaeb0cc62c8347ed779a5caa45e714071/jans-auth-server/server/src/main/webapp/WEB-INF/incl/layout/authorize-extended-template.xhtml#L61

And description of spontaneous scope is `"Spontaneous scope: " + <scope object address>`.
This indeed is not very useful. I've just raised PR to put `scopeId` in description 
https://github.com/JanssenProject/jans/pull/9752

It will be available in next release. If you can't wait you can workaround with at least two ways:

1. provide own customized authorization page and render in the way it is needed instead of 
```
                                                       <ui:repeat var="_scope" value="#{_scopes}">
								<li>
									<table>
										<tr>
											<td><span class="user"><i
													class="fa fa-user-circle"></i></span></td>
											<td><h:outputText value="#{_scope.description}" /></td>
										</tr>
									</table>
								</li>
							</ui:repeat>
```

2. second way it to overwrite description in some custom script. For example in `PostAuthnType` in any method (or some other script).
You can identify all scopes by `creatorId`. If it's client's scopes then simply `creatorId=<clientId>`.

**Comment by tawaren** [comment]
Thanks. I can work with the workarounds if I need it before the next release happens.
